cmd/proxy/Dockerfile

```diff
@@ -1,13 +1,30 @@
-FROM golang:1.11-rc
+FROM gobuffalo/buffalo:v0.12.6 as buffalo-builder
+
+RUN mkdir -p $GOPATH/src/github.com/gomods/athens/cmd/proxy
+WORKDIR $GOPATH/src/github.com/gomods/athens/cmd/proxy
+
+# this will cache the npm install step, unless package.json changes
+ADD cmd/proxy/package.json .
```

Comment:
Why do we need yarn, package.json?

We do not have a web app for MVC.

Reply:
MVP**

Reply:
I see, we don't want any HTML/JS assets in either the proxy or Olympus? Do we still want to serve a basic HTML page from Go for verification purposes?

If we don't need HTML/JS, perhaps we can use `go build` instead of `buffalo build` - that's what @marwan-at-work was proposing I think.

Reply:
Right! We can always hit the `healthz` endpoint to verify if everything is working

https://github.com/gomods/athens/blob/master/cmd/proxy/actions/health.go
https://github.com/gomods/athens/blob/d749465e35e5183cde0c1cfd89327ca3f1eadca6/cmd/proxy/actions/app_proxy.go#L18

As to the second part, the routers are still part of the `buffalo build`. There is a PR that 
removes pop dependency from buffalo build in proxy and allows buffalo build and buffalo test to run without setting up a database
https://github.com/gomods/athens/pull/511

I am not sure if we can really do go build as of now.